demos/python_demos/handwritten_japanese_recognition_demo/utils/codec.py

```diff
@@ -3,8 +3,9 @@
 
 class CTCCodec(object):
     """ Convert between text-label and text-index """
-    def __init__(self, characters):
+    def __init__(self, characters, args):
```

Comment:
Since you use only `designated_characters` and `top_k` values, you should cut `args` to only these values. Or may be even move to `decode` method.